src/search/airportsearch.cpp

```diff
@@ -744,14 +750,166 @@ void AirportSearch::randomFlightplanClicked()
 {
   Ui::MainWindow *ui = NavApp::getMainUi();
 
+  int distanceMin = ui->spinBoxAirportFlightplanMinSearch->value();
+  int distanceMax = ui->spinBoxAirportFlightplanMaxSearch->value();
+
   // Log minimum and maximum distance from UI
-  qDebug() << Q_FUNC_INFO << "min" << ui->spinBoxAirportFlightplanMinSearch->value()
-           << "max" << ui->spinBoxAirportFlightplanMaxSearch->value();
+  qDebug() << Q_FUNC_INFO << "random flight, distance min: " << distanceMin
+           << ", random flight, distance max: " << distanceMax;
+
+  if(distanceMin < distanceMax) {
+    // TODO:
+    // non-blocking notification: the closer the distance (max - min) is to 0, the longer the random selection might take
+    // give estimated search value: ((20500 - (max - min)) / 20500) * 100% of all airports; 20500 is current max of distanceMax
+    // TODO:
+    // if estimated search value is > 99,5% : ask user if he we wants to let continue ( > 99,5% == < 100 km distance)
+    // NOTE:
+    // currently no interrupt after time or x attempts
```

Comment:
Simply remember start time from `QDateTime::current...()` and compare to current time. Do this only every N-th iterations since even fetching time takes time. I'd always stop searching after five seconds or whatever the typical attention span of a user is. :upside_down_face:

Reply:
I implemented this and tried to be clever: I ask whether the user likes to continue, every 10 seconds. The longest I ever waited on multiple attempts to 20000 to 20010 was 25 seconds. Non-Modal boxes don't get rendered unfortunately (because Qt is budy calculating, but I could move it to a thread) thus I have a modal with 2 buttons (Yes, no, rather than just "abort" and keeping it displayed while it's running).

Reply:
I see. Yours is the better solution. :+1: 

You can call `QApplication::processEvents()` to get the dialog rendered. I had a lot of issues with this too and it also affects the display of progress dialogs. I'd use a modal dialog since there can be unwanted user input.

The problem is that the program processes input too while calling `processEvents()`. Gotta be cautious with this.

You can also set a wait cursor to let the user know that program is doing something:
```
   QGuiApplication::setOverrideCursor(Qt::WaitCursor);
   // ...
   QGuiApplication::restoreOverrideCursor();
```

Note about process events: Some OS consider the program frozen/crashed if this is not called for a longer time.

Reply:
Here's an example for a progress dialog. This one appears automatically if the calculation takes more time.
No need to show real progress. Simply update with the same value or do a rough estimate.

https://github.com/albar965/littlenavmap/blob/e64061b8f9e9ac1e11952e5a4257cafa2803edf5/src/route/routecontroller.cpp#L1734-L1761

Reply:
thanks, I will try a few more things here. But might be around 7 pm when i push something.

Reply:
Take your time. No prob.

Reply:
I finally got progressbar to work. According to https://doc.qt.io/qt-5/qprogressdialog.html#value-prop I needed to set the minimum first and actually use a non modal dialog. The modal one didn't work (and would have called `processEvents` itself)
I also added multi-threading. Uses 1 thread per logical core. Process Lasso cpu affinity did put all created threads to the core lnm was designated to thus such should be disabled. Works on a per departure basis. Ie. the algorithm took a departure and compared all possible airports as destinations. If no sucess it took another departure and continued. Now those departures are run in parallel (but thus each thread possibly processing all other ~ 37 000 airports as comparison for a destination). This can only be a benefit if the original random departure was unsuited for the distance criteria. Actually parallelising the destination search would be complex.

Reply:
> I finally got progressbar to work. According to https://doc.qt.io/qt-5/qprogressdialog.html#value-prop I needed to set the minimum first and actually use a non modal dialog. The modal one didn't work (and would have called `processEvents` itself)

The modal dialog calls process events which allows some background activity like aircraft position update (you can still see your aircraft moving while loading the scenery library) put it forbids user input which is the important part.